src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -554,5 +555,85 @@ public enum SampleEnumSByte : sbyte
             F = 1 << 5,
             G = 1 << 6,
         }
+
+        [Fact]
+        public static void Honor_EnumNamingPolicy_On_Deserialization()
+        {
+            JsonSerializerOptions options = new()
+            {
+                Converters =
+                {
+                    new JsonStringEnumConverter(namingPolicy: new SimpleSnakeCasePolicy() )
+                }
+            };
+
+            BindingFlags bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""non_public""", options);
+            Assert.Equal(BindingFlags.NonPublic, bindingFlags);
+
+            // Flags supported without naming policy.
+            bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""NonPublic, Public""", options);
+            Assert.Equal(BindingFlags.NonPublic | BindingFlags.Public, bindingFlags);
+
+            // Flags supported with naming policy.
+            bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""static, public""", options);
+            Assert.Equal(BindingFlags.Static | BindingFlags.Public, bindingFlags);
+
+            // Null not supported.
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<BindingFlags>("null", options));
+
+            // Null supported for nullable enum.
+            Assert.Null(JsonSerializer.Deserialize<BindingFlags?>("null", options));
+        }
+
+        [Fact]
+        public static void EnumDictionaryKeyDeserialization()
```

Comment:
I'd just joing those two tests into one since they test the same concept and are mostly the same